kitsune/wiki/utils.py

```diff
@@ -178,7 +179,42 @@ def get_featured_articles(product=None, locale=settings.WIKI_DEFAULT_LANGUAGE):
 
 
 def get_visible_document_or_404(user, **kwargs):
-    return get_object_or_404(Document.objects.visible(user, **kwargs))
+    """
+    Get the document specified by the keyword arguments and visible to the given user, or 404.
+    """
+
+    look_for_translation_via_parent = kwargs.pop("look_for_translation_via_parent", False)
+    return_parent_if_no_translation = kwargs.pop("return_parent_if_no_translation", False)
+
+    try:
+        return Document.objects.get_visible(user, **kwargs)
+    except Document.DoesNotExist:
+        pass
+
+    if (
+        not look_for_translation_via_parent
+        or not (locale := kwargs.get("locale"))
```

Comment:
I think that would break the logic, since I'm checking that either `locale` is not provided or it's set to the value of `settings.WIKI_DEFAULT_LANGUAGE`. If I collapsed the lines, the case when `locale` is not provided would never trigger.

Reply:
If it's not provided, it would default to an empty string which will not be equal to the default locale. In case it is provided, it will check against the default which is the second part. This is very nit. I am just curious about it to figure out what I am missing

Reply:
> If it's not provided, it would default to an empty string which will not be equal to the default locale.

That wouldn't match the current logic, since it would be `False` and not trigger the clause and raise an Http404 exception. So to match the logic, it would have to instead be:

```python
kwargs.get("locale", settings.WIKI_DEFAULT_LANGUAGE) == settings.WIKI_DEFAULT_LANGUAGE
```

so that's why I kept it as I had it.